### ISSUE TITLE
Llama‘s freqs_cos data loss as for convert dtype

### ISSUE THREAD

**Issue by WeiMa01** [pytorch/executorch#9393]
- The freqs_cos of the Llama model structure of ExecuTorch is FP32 at the time of its creation
calculation of freqs_cos by precompute_freqs_cis：
` freqs = 1.0 / (
        theta ** (torch.arange(0, dim, 2, device="cpu")[: (dim // 2)].float() / dim)
    )`

- The model is loaded by get_eager_model and is converted from FP32 to BF16 (Llama original model data type). This can lead to data loss and affect inferences
`    def get_eager_model(self) -> torch.nn.Module:

        if self.dtype:
            return self.model_.to(self.dtype)`

- I think freqs_cos should always keep FP32

**Comment by kimishpatel** [comment]
Makes sense. Does this happen when you override dtype during model export via `-d bf16`?

**Comment by kimishpatel** [comment]
cc @jackzhxng

**Comment by jackzhxng** [comment]
@WeiMa01 Seems like the issue I was trying to fix in this PR: https://github.com/pytorch/executorch/pull/8441 a while back. I close that PR in favor of this one - https://github.com/pytorch/executorch/pull/8488, which I am trying to land today. This should address your problem

**Comment by WeiMa01** [comment]
@jackzhxng ，thank you， it seems that you remove dtype converting of get_eager_model. it can solve my issue
